packages/shared/createGlobalState/index.md

```diff
@@ -11,19 +11,24 @@ Keep states in the global scope to be reusable across Vue instances.
 
 ### Without Persistence (Store in Memory)
 
-```js
-// store.js
+<div class="ts-api">
```

Comment:
Can we just automatically apply this to any ts block, without the need of this marker? (so the documents keep untouched)

Reply:
the `<div class="ts-api">...</div>` and `<div class="js-api">...</div>` are identifier classes for when to hide which code block when the toggle is used.

Please refer the below code in `TsJsToggleSwitch.vue`:

https://github.com/vueuse/vueuse/pull/3082/files#diff-8ac7c8298153b8b006a414a6ff0abde634bd9a6483eec29d42ad4387981aa170R174-R184

What `TsJsToggleSwitch.vue` does is, the default code enabled is the Typescript code, when we toggle to Javascript, the component adds a `prefer-js` class to the body. Once this class is attached to the body, the typescript code (one with class `ts-api`) gets a `display: none;` css property and the Javascript code (one with class `js-api`) gets a `display: initial;` css property. Similarly, when you toggle it back to typescript, the Javascript code gets removed from the DOM and the Typescript code gets added back to the DOM. Hence the `div` elements in the documents.

I have also written a vite plugin `tsToJsTransform.ts`, which recognizes the typescript code using the divs with the `ts-api` and transpiles the code inside it to Javascript code and adds both the snippets in the file code at build time. Correct me if I am wrong, but we are supposed to rewrite all the JS snippets as TS snippets and transpile those snippets on build time to generate an additional JS code block.

If you have a better approach, please let me know, and I shall try to go that way.

Reply:
@antfu Please let me know if you have a better approach...

Reply:
@antfu Can you please update on my comment?